src/storage/kv/transaction.rs

```diff
@@ -157,7 +158,11 @@ impl Transaction {
                 // Check if the key is in the write set by checking in the write_order_map map.
                 if let Some(order) = self.write_order_map.get(&hashed_key) {
                     if let Some(entry) = self.write_set.get(*order as usize) {
-                        return Ok(Some(entry.1.value.clone().to_vec()));
+                        return Ok(if entry.1.is_deleted() {
```

Comment:
@mumoshu This is actually the place. So when an entry is deleted, I store empty value against it in the snapshot, but when I was reading it, it used to return Some([]) when it had to return None.